docs/contracts/visibility-and-getters.rst

```diff
@@ -152,44 +150,44 @@ it evaluates to a state variable.  If it is accessed externally
     contract C {
         uint public data;
         function x() public returns (uint) {
-            data = 3; // internal access
-            return this.data(); // external access
+            data = 3; // internal erişim
+            return this.data(); // external erişim
         }
     }
 
-If you have a ``public`` state variable of array type, then you can only retrieve
-single elements of the array via the generated getter function. This mechanism
-exists to avoid high gas costs when returning an entire array. You can use
-arguments to specify which individual element to return, for example
-``myArray(0)``. If you want to return an entire array in one call, then you need
-to write a function, for example:
+Eğer bir ``public`` görünürlüğe sahip dizi tipinden bir durum değişkenine sahipseniz, getter
+fonksiyonunu kullanarak sadece tek bir elemana erişim sağlayabilirsiniz. Bu mekanizma
+tüm diziyi return ederken oluşan yüksek gaz ücretlerinden sıyrılmak için kurulmuştur. Hangi elemanın
+return edileceğini belirtmek için parametreleri kullanabilirsiniz (örneğin, ``myArray(0)``).
+Eğer bütün diziyi tek bir fonksiyon ile elde etmeniz gerekiyorsa, bunun için aşağıdaki gibi
+bir fonksiyon yazmanız gerekir.
 
 .. code-block:: solidity
 
     // SPDX-License-Identifier: GPL-3.0
     pragma solidity >=0.4.16 <0.9.0;
 
     contract arrayExample {
-        // public state variable
+        // public durum değişkeni
         uint[] public myArray;
 
-        // Getter function generated by the compiler
+        // Derleyici tarafından tanımlanan getter fonksiyonu
         /*
         function myArray(uint i) public view returns (uint) {
             return myArray[i];
         }
         */
 
-        // function that returns entire array
+        // Bütün array'i return eden fonksiyon
         function getArray() public view returns (uint[] memory) {
             return myArray;
         }
     }
 
-Now you can use ``getArray()`` to retrieve the entire array, instead of
-``myArray(i)``, which returns a single element per call.
+Şimdi ``getArray()`` fonksiyonunu kullanarak bütün array'i elde edebilirsiniz. Tek tek bütün elemanları
+``myArray(i)`` kullanarak çağırmanıza gerek kalmadı.
```

Comment:
'kullanarak çağırmanıza gerek kalmadı' kısmı kulağa hoş gelmiyor.  şu şekilde değiştirilebilir:
Artık tüm dizine erişmek için her bir aramayı  tek bir öğeye döndüren ``myArray(i)`` yerine ``getArray()`` kullanabilirsiniz.